python/cudf/cudf/_libxx/nvtx.pyx

```diff
@@ -25,38 +38,14 @@ def range_push(object name, object color='green'):
     """
     cdef const char* _name = name
     cdef uint32_t _color = color
+    cdef color_type _Color = Color[color]
     try:
         _color = int(_color, 16)
         with nogil:
             cpp_range_push_hex(_name, _color)
     except ValueError:
-        if color == 'green':
-            with nogil:
-                cpp_range_push(_name, nvtx_color.GREEN)
-        elif color == 'blue':
-            with nogil:
-                cpp_range_push(_name, nvtx_color.BLUE)
-        elif color == 'yellow':
-            with nogil:
-                cpp_range_push(_name, nvtx_color.YELLOW)
-        elif color == 'purple':
-            with nogil:
-                cpp_range_push(_name, nvtx_color.PURPLE)
-        elif color == 'cyan':
-            with nogil:
-                cpp_range_push(_name, nvtx_color.CYAN)
-        elif color == 'red':
-            with nogil:
-                cpp_range_push(_name, nvtx_color.RED)
-        elif color == 'white':
-            with nogil:
-                cpp_range_push(_name, nvtx_color.WHITE)
-        elif color == 'darkgreen':
-            with nogil:
-                cpp_range_push(_name, nvtx_color.DARK_GREEN)
-        elif color == 'orange':
-            with nogil:
-                cpp_range_push(_name, nvtx_color.ORANGE)
+        with nogil:
+            cpp_range_push(_name, _Color)
```

Comment:
Can we catch this before we may the `cpp` call. If the name isn't in the enum we shouldn't call the `cpp_range_push_hex` regardless.